blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeTest.java

```diff
@@ -507,29 +512,34 @@ public void testNestedBlockStage() throws Exception{
                 Assert.assertEquals(1, edges.size());
                 Assert.assertEquals(rn.get("result"), "SUCCESS");
                 Assert.assertEquals(rn.get("state"), "FINISHED");
-            }else if(rn.get("displayName").equals("test")){
+            }else if(rn.get("displayName").equals("Packaging")){
```

Comment:
🐛 We do not support nested stages in scripted pipeline. Only sequential stages should appear in declarative pipeline. 

@abayer Assuming we render nested stages on in case of declarative sequential stage: is there way to find we are executing declarative pipeline vs scripted? Maybe from `WorkFlowRun` or `FlowExecution`? Or maybe from `FlowNode` corresponding to a stage being inspected while building graph in API call to exclude nested stages in scripted but show them in the declarative context for sequential stages?

Reply:
`WorkflowRun#getAction(ExecutionModelAction.class)` - `ExecutionModelAction` gets attached to every Declarative run, and won't be on any Scripted run. What's more, `ExecutedModelAction` actually contains the execution plan for the Declarative Pipeline, via [`#getStages()`](https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/master/pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/actions/ExecutionModelAction.java#L62), which will give you the [`ModelASTStages`](https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/master/pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTStages.java) that will be/are being executed by the run.